src/EventStore.Core/DataStructures/ProbabilisticFilter/AlignedMemory.cs

```diff
@@ -42,8 +42,12 @@ public sealed unsafe class AlignedMemory : IDisposable {
 
 			_disposed = true;
 			GC.SuppressFinalize(this);
-			Marshal.FreeHGlobal(_intPtr);
-			GC.RemoveMemoryPressure(_bytesAllocated);
+
+			if (_intPtr != IntPtr.Zero) {
+				// didn't manage to allocate or add pressure (OOM), so do not free.
```

Comment:
```suggestion
				// didn't manage to allocate or add pressure (OOM), so do not free.
```
shouldn't this comment be in the `else` block of this `if`?

Reply:
oops, thanks. fixed

Reply:
thanks!